core/backend/src/IModelDb.ts

```diff
@@ -606,6 +608,32 @@ export abstract class IModelDb extends IModel {
     return stmt;
   }
 
+  /**
+   * queries the BisCore.SubCategory table for the entries that are children of the passed categoryIds
+   * @param categoryIds categoryIds to query
+   * @returns array of SubCategoryResultRow
+   */
+  public async querySubCategories(categoryIds: Id64String[]): Promise<SubCategoryResultRow[]> {
+    // consider splitting up categoryIds, as queries get slow with many many categoryids in them.
+    const maxCategoriesPerQuery = 200;
+    const result = [];
+    while (categoryIds.length !== 0) {
+      const end = (categoryIds.length > maxCategoriesPerQuery) ? maxCategoriesPerQuery : categoryIds.length;
+      const where = categoryIds.splice(0, end).join(",");
+      const query = `SELECT ECInstanceId as id, Parent.Id as parentId, Properties as appearance FROM BisCore.SubCategory WHERE Parent.Id IN (${where})`;
+      try {
+        for await (const row of this.query(query, undefined, { rowFormat: QueryRowFormat.UseJsPropertyNames })) {
+          result.push(row);
+        }
+      } catch {
+        // ###TODO: detect cases in which retry is warranted
```

Comment:
This comment copied from frontend is not meaningful in backend.

Reply:
changed it